confdb/aclocal_modules.m4

```diff
@@ -54,28 +54,39 @@ AC_DEFUN([PAC_CONFIG_HWLOC_EMBEDDED],[
 ])
 
 AC_DEFUN([PAC_CONFIG_HWLOC],[
+    dnl hydra inherits CFLAGS from mpich. When there is hwloc v1 installed, it may result in
+    dnl compiling with embedded hwloc headers but linking with (incompatible) external hwloc.
+    dnl Check to prevent this case.
+    if test "$FROM_MPICH" = "yes" ; then
+        if echo "$CPPFLAGS" | grep -q "modules/hwloc" ; then
+            with_hwloc=embedded
+        fi
+    fi
```

Comment:
At this point in the MPICH configure, `"$FROM_MPICH" = "yes"`, but it shouldn't mess up the logic.

A separate more general question is, should Hydra inherit CFLAGS from MPICH? Obviously it should inherit USER_CFLAGS and anything we specify in PAC_CONFIG_SUBDIRS. We can think about that separately, since its a longstanding behavior.

Reply:
Sanitizing CFLAGS/CPPFLAGS seems to be a cleaner solution. Let me try that.

Reply:
Testing the idea of reset all flags. Will squash away the extra commits later.

Reply:
done.